### PR TITLE
Warn the user when closing/reloading page if the report is being saved

### LINKED ISSUE #127: add beforeunload handler when saving the report

### PR DESCRIPTION
**Pull request by wcandillon**
fixes #127

### CODE REVIEW

**Comment by dknochen** [comment]
@wcandillon it looks good. Could we add a test for it?

File: app/modules/ui/api-status.js [hunk 1]
```diff
@@ -2,7 +2,22 @@
 
 angular
 .module('report-editor')
-.directive('apiStatus', function($rootScope, $timeout){
+.directive('apiStatus', function($rootScope, $timeout, $window){
+    var code;
+    var onBeforeUnloadHandler = function(event){
+        if(code === 'warning') {
```
Comment (wcandillon) [inline app/modules/ui/api-status.js:8]: @dknochen Removing the if causes the test suite to regress. This means that from no one any illegal actions in the test suite are caught

**Comment by wcandillon** [comment]
@dknochen The fact that the test suite pass with this change is already 50% of the testing for this feature. I tried to caught the alert while the report is saving but I am blocked on this issue:  https://github.com/angular/protractor/issues/308

File: tests/e2e/saving-scenario_.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    var id = '1fueA5hrxIHxvRf7Btr_J6efDJ3qp-s9KV731wDc4OOaw';
+    var Report = require('../../app/report/report-page');
+    var FiltersPage = require('../../app/report/filters/filters-page');
+
+    it('Should throw an exception if we reloading while saving the report', function(){
+        var report = new Report(id);
+        var filters = report.filters;
+        filters.visitPage();
+        //Let's do an illegal update where we don't wait for angularjs to finish
```
Comment (wcandillon) [inline tests/e2e/saving-scenario_.js:14]: @dknochen that would be the test that I am not sure how to finish yet.

Comment (dknochen) [inline tests/e2e/saving-scenario_.js:14]: @wcandillon mmmh. What about broadcasting an apiStatus event to set the code manually and then closing the window? http://stackoverflow.com/questions/23544106/can-protractor-be-used-to-test-multiple-browser-session-interactions

**Pull request merged**